src/marqo/instance_mappings.py

```diff
@@ -35,3 +35,14 @@ def get_control_base_url(self) -> str:
     @abstractmethod
     def is_remote(self):
         pass
+
+    @abstractmethod
+    def on_instance_error(self, index_name: str, http_status: int) -> None:
```

Comment:
as in on a specific Marqo dataplane? We can probably call this `index_http_error_handler` to be more descriptive. Is it OK to use "index" over "instance" - I think the abstractions still hold.